test/unit/BondingManager.js

```diff
@@ -446,6 +446,29 @@ contract("BondingManager", accounts => {
                 )
             })
 
+            it("fires an EarningsClaimed event when bonding from unbonded", async () => {
+                const txResult = await bondingManager.bond(1000, transcoder0, {from: delegator})
+
+                truffleAssert.eventEmitted(
+                    txResult,
+                    "EarningsClaimed",
+                    e => e.delegate === constants.NULL_ADDRESS &&
+                        e.delegator == delegator &&
+                        e.fees == 0 &&
+                        e.rewards == 0 &&
+                        e.startRound == 1 &&
+                        e.endRound == currentRound,
+                    "EarningsClaimed event not emitted correctly"
+                )
+            })
+
+            it("it doesn't fire an EarningsClaimed event when bonding twice in the same round", async () => {
+                await bondingManager.bond(1000, transcoder0, {from: delegator})
+                const txResult = await bondingManager.bond(1000, transcoder0, {from: delegator})
+
+                txResult.logs.forEach(l => assert.notEqual(l.event, "EarningsClaimed", "Logs should not include an EarningsClaimed event"))
```

Comment:
Can we use [truffleAssert.eventNotEmitted()](https://github.com/rkalis/truffle-assertions#truffleasserteventnotemittedresult-eventtype-filter-message) here since we use `truffleAssert` elsewhere for event related assertions?

Reply:
Ah yes we can, I didn't bother checking the truffleAssert API for such a method, perhaps should have.

Reply:
fixed in 6e24b69